sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/ArraySqlAggregator.java

```diff
@@ -114,7 +113,7 @@ public Aggregation toDruidAggregation(
 
     final String fieldName;
     final String initialvalue;
-    final ColumnType druidType = Calcites.getValueTypeForRelDataTypeFull(aggregateCall.getType());
+    final ColumnType druidType = Calcites.getColumnTypeForRelDataType(aggregateCall.getType());
```

Comment:
we should maybe still call the old version of this method so that behavior doesn't change when the flag is set to the old behavior (which would change the behavior of this method)

Reply:
good catch. Fixed.